File: src/data/types/shipComponent.ts
import type { ComponentCategory } from "./componentCategory.ts";
import type { QualityRoll } from "./quality.ts";
import type { TierColor } from "./tierColor.ts";

// Phase 5 GDD §2.1/§2.2/§2.5 -- a crafted ship component. Carries the same
// 5-quality data as every other crafted item (no separate stat system);
// `tier` is derived from those qualities via the existing straight-
// average-to-tier aggregation (src/simulation/aggregateTier.ts), same
// formula already used for crafted-item display tier and Listing.marketTier.
export interface ShipComponent {
  id: string;
  category: ComponentCategory;
  qualities: QualityRoll;
  tier: TierColor;
}

File: unity/ProfitableCore/Schema/ShipComponent.cs
namespace Profitable.Core.Schema;

// Ports src/data/types/shipComponent.ts. Carries the same 5-quality data
// as every other crafted item; Tier is derived from those qualities via
// AggregateTierResolver, same formula already used for Listing.MarketTier.
public class ShipComponent
{
    public string Id { get; set; } = string.Empty;
    public ComponentCategory Category { get; set; }
    public QualityMap Qualities { get; set; } = new();
    public TierColor Tier { get; set; }
}
